components/admin_console/billing/billing_subscriptions.tsx

```diff
@@ -53,14 +53,15 @@ const BillingSubscriptions: React.FC<Props> = () => {
     const currentUser = useSelector((state: GlobalState) => getCurrentUser(state));
     const isCloud = useSelector((state: GlobalState) => getLicense(state).Cloud === 'true');
     const subscription = useSelector((state: GlobalState) => state.entities.cloud.subscription);
-    const customer = useSelector((state: GlobalState) => state.entities.cloud.customer);
+
     const products = useSelector((state: GlobalState) => state.entities.cloud.products);
+    const isCardExpired = useSelector((state: GlobalState) => isCustomerCardExpired(state.entities.cloud.customer));
     const getCategory = makeGetCategory();
     const preferences = useSelector<GlobalState, PreferenceType[]>((state) => getCategory(state, Preferences.ADMIN_CLOUD_UPGRADE_PANEL));
 
     const contactSalesLink = useSelector((state: GlobalState) => getCloudContactUsLink(state, InquiryType.Sales));
 
-    const [showCreditCardBanner, setShowCreditCardBanner] = useState(isCustomerCardExpired(customer));
```

Comment:
Commenting here to explain the PR @marianunez (did a little more research into when hooks are called). This line is the problem: `const [showCreditCardBanner, setShowCreditCardBanner] = useState(isCustomerCardExpired(customer));` is equivalent to (more or less)
```
constructor(props) {
  this.state = {showCreditCardBanner: isCustomerCardExpired(customer)}
}
```
Meaning that once the state variable is initialized, subsequent re-renders (like the re-render that happens when the customer entity is updated) wouldn't actually change the state value, because the setting of the initial state ( the call to `isCustomerCardExpired(customer)`) is only run once.

Reply:
Thank you for the detailed explanation, makes sense to me now!